Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -65,9 +65,53 @@ public class Mqtt5Client {
         // TODO Error should be thrown or client nil logged
     }
 
+    /// Tells the client to attempt to subscribe to one or more topic filters.
+    ///
+    /// - Parameters:
+    ///     - subscribePacket: SUBSCRIBE packet to send to the server
+    /// - Returns:
+    ///     - @return Suback packet if the subscription operation succeed otherwise errorCode
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func subscribe(subscribePacket: SubscribePacket) async throws -> SubackPacket {
+
+        return try await withCheckedThrowingContinuation { continuation in
+            subscribePacket.withCPointer { subscribePacketPointer in
+                var callbackOptions = aws_mqtt5_subscribe_completion_options()
+                let continuationCore = ContinuationCore(continuation: continuation)
+                callbackOptions.completion_callback = subscribeCompletionCallback
+                callbackOptions.completion_user_data = continuationCore.passRetained()
+                let result = aws_mqtt5_client_subscribe(rawValue, subscribePacketPointer, &callbackOptions)
+                guard result == AWS_OP_SUCCESS else {
+                    continuationCore.release()
+                    return continuation.resume(throwing: CommonRunTimeError.crtError(CRTError.makeFromLastError()))
+                }
+            }
+
+        }
+    }
+
     public func close() {
         self.callbackCore.close()
         aws_mqtt5_client_release(rawValue)
         rawValue = nil
     }
 }
+
+// The completion callback to invoke when an ack is received in native
+private func subscribeCompletionCallback(
```

Comment:
Does this work? This callback doesn't seem to fit the native callback signature. (packet_type, packet, error_code, context)